docs/en/deploy/Running-Alluxio-Using-Vagrant.md

```diff
@@ -5,41 +5,127 @@ nickname: Alluxio on Vagrant
 group: Deploying Alluxio
 priority: 5
 ---
-TODO(david): Outline
-  - Table of Contents
-  - Introduction
-  - Prerequisite for using Vagrant
-  - Deploy using Vagrant on Virtual Box
-    - Launch Cluster
-    - Access / Verify the Cluster
-    - Destroy 
-  - Deploy using Vagrant on EC2
-    - Launch Cluster
-    - Access / Verify the Cluster
-    - Destroy
-    - Advanced Tips
-      -Spot Instance
-  - Deploy using Vagrant on GCE
-    - Launch Cluster
-    - Access / Verify the Cluster
-    - Destroy
-
-TODO(david): combine content from Deploy on EC2, GCE, Virtual Box using Vagrant 
-
 * Table of Contents
 {:toc}
 
-Alluxio can be deployed on Amazon EC2 using the [Vagrant scripts](https://github.com/alluxio/alluxio/tree/master/deploy/vagrant) that come with Alluxio. The scripts let you create, configure, and destroy clusters that come automatically configured with [Amazon S3](https://s3.amazonaws.com/).
+## Introduction
+Alluxio can be deployed locally or in the cloud using the [Vagrant scripts](https://github.com/alluxio/alluxio/tree/master/deploy/vagrant) that come with Alluxio. The scripts let you create, configure, and destroy clusters that come automatically configured locally, with [Amazon EC2](https://ec2.amazon.com/) or with [Google Compute Engine](https:/cloud.google.com).
+
+## Deploy using Vagrant on Virtual Box
+### Prerequisites
+**Install VirtualBox**
+
+Download [VirtualBox](https://www.virtualbox.org/wiki/Downloads)
+
+**Install Vagrant**
+
+Download [Vagrant](https://www.vagrantup.com/downloads.html)
+
+**Install Alluxio**
+
+[Clone Alluxio Repository](https://github.com/Alluxio/alluxio) to your local machine
+
+**Install python library dependencies**
+
+Install [python>=2.7](https://www.python.org/), not python3.
+
+Under `deploy/vagrant` directory in your home directory, run:
+
+```bash
+$ sudo bash bin/install.sh
+```
+
+Alternatively, you can manually install [pip](https://pip.pypa.io/en/latest/installing/), and then
+in `deploy/vagrant` run:
+
+```bash
+$ sudo pip install -r pip-req.txt
+```
+### Launch Cluster
+
+Now you can launch the Alluxio cluster with Hadoop2.4.1 as under filesystem by running the script under deploy/vagrant:
+
+```bash
+$ ./create <number of machines> vb
+```
+
+Each node of the cluster runs an Alluxio worker, and the AlluxioMaster runs an Alluxio master.
+
+### Access and Verify Cluster
```

Comment:
will this section repeat multiple times per deploy destination?

Reply:
This section is slightly different for each deploy destination... that is what makes it a bit tricky.. I can remove all the deployment specific part and have one common section.